src/puptoo/process/profile.py

```diff
@@ -261,16 +261,18 @@ def system_profile(
             if sap.local_instances:
                 inst = sap.local_instances[0]
                 profile["sap_instance_number"] = sap[inst].number
-            sap_object = {
-                "sap_system": profile.get("sap_system"),
-                "sids": profile.get("sap_sids"),
-                "instance_number": profile.get("sap_instance_number")
-            }
-            profile["sap"] = sap_object
+            profile["sap"] = {}
+            profile["sap"]["sap_system"] = profile.get("sap_system")
+            if profile.get("sap_sids") and profile.get("sap_instance_number"): 
+                sap_object = {
+                    "sids": profile.get("sap_sids"),
+                    "instance_number": profile.get("sap_instance_number")
+                }
+                profile["sap"] = sap_object
```

Comment:
@r14chandra  - Could you please confirm here expected behavior is to append/replace the existing profile["sap"] object?